src/Build/Logging/BinaryLogger/BinaryLogger.cs

```diff
@@ -80,6 +80,8 @@ public sealed class BinaryLogger : ILogger
         //    BuildCheckTracingEvent, BuildCheckAcquisitionEvent, BuildSubmissionStartedEvent
         // version 24:
         //    - new record kind: BuildCanceledEventArgs
+        // version 25:
```

Comment:
Hmm, do we even need to increment the file format version any more?

I'm not 100% but I don't think so. Since the reader will correctly read the Message, File, Line and Column whether they are present or not (it will read the Flags and then read the fields based on the flags).

We should think about two scenarios:
 1. new reader reading an older binlog (written before this PR)
 2. old reader reading a newer binlog (written after this PR)

If both of these scenarios work fine, then we don't need to increment the version.

Reply:
I think I still need this version increase due to a different way  of formatting location.
Previously, I was passed as a LocationString, now I format this on the receiving side:
https://github.com/KirillOsenkov/MSBuildStructuredLog/blob/9bc29f2684c57a5424827d590f8b180c46905c89/src/StructuredLogger/BinaryLogger/BuildEventArgsReader.cs#L1430

or on message read
https://github.com/dotnet/msbuild/blob/48107543350750cc1faab0f9965602c1051a81ce/src/Framework/UninitializedPropertyReadEventArgs.cs#L62

Reply:
OK then always better to increment the version to be safe